addons/hr_expense/models/hr_expense.py

```diff
@@ -313,10 +313,12 @@ def _compute_state(self):
 
     @api.depends('quantity', 'price_unit', 'tax_ids')
     def _compute_total_amount_currency(self):
+        AccountTax = self.env['account.tax']
         for expense in self.filtered('product_has_cost'):
-            base_lines = [expense._convert_to_tax_base_line_dict(price_unit=expense.price_unit, quantity=expense.quantity)]
-            taxes_totals = self.env['account.tax']._compute_taxes(base_lines, expense.company_id)['totals'][expense.currency_id]
-            expense.total_amount_currency = taxes_totals['amount_untaxed'] + taxes_totals['amount_tax']
+            base_line = expense._prepare_base_line_for_taxes_computation(price_unit=expense.price_unit, quantity=expense.quantity)
+            AccountTax._add_base_line_tax_details(base_line, expense.company_id)
+            AccountTax._round_base_lines_tax_details([base_line], expense.company_id)
```

Comment:
aren't we always calling `_add_base_line_tax_details` then `_round_base_lines_tax_details` ?  is there a use case where we could call one without the other?

maybe introducing a method that does that would clean/simplify the understanding one could have? :thinking: 

edit: maybe even the 3 methods are always together?

Reply:
Yeah most of the time... The problem is sometimes, you want to call `_prepare_base_line_for_taxes_computation`, then change the dictionary before calling `_add_base_line_tax_details`.

Reply:
well in those (rare?) cases we could still call the inner methods separately. But here honestly it makes the whole code complex to understand

Reply:
added a TODO for post OXP